cli/commands/dream/build/function.go

```diff
@@ -0,0 +1,51 @@
+package build
+
+import (
+	"path"
+
+	commonSpec "github.com/taubyte/go-specs/common"
+	functionSpec "github.com/taubyte/go-specs/function"
+	dreamI18n "github.com/taubyte/tau/i18n/dream"
+	applicationLib "github.com/taubyte/tau/lib/application"
+	dreamLib "github.com/taubyte/tau/lib/dream"
+	functionPrompts "github.com/taubyte/tau/prompts/function"
+	"github.com/urfave/cli/v2"
+)
+
+func buildFunction(ctx *cli.Context) error {
+	if !dreamLib.IsRunning() {
+		dreamI18n.Help().IsDreamlandRunning()
+		return dreamI18n.DreamlandNotStarted
+	}
+
+	function, err := functionPrompts.GetOrSelect(ctx)
+	if err != nil {
+		return err
+	}
+
+	builder, err := initBuild()
+	if err != nil {
+		return err
+	}
+
+	compileFor := &dreamLib.CompileForDFunc{
+		ProjectId:  builder.project.Get().Id(),
+		ResourceId: function.Id,
+		Branch:     builder.currentBranch,
+		Call:       function.Call,
+	}
+
+	if len(builder.selectedApp) > 0 {
+		app, err := applicationLib.Get(builder.selectedApp)
+		if err != nil {
+			return err
+		}
+
+		compileFor.ApplicationId = app.Id
+		compileFor.Path = path.Join(builder.projectConfig.CodeLoc(), commonSpec.ApplicationPathVariable.String(), builder.selectedApp, functionSpec.PathVariable.String(), function.Name)
```

Comment:
Not relevant to this PR, but in general I feel like `Loc` used for shorthand for Location sucks. When I read Loc, I process Lock initially. Loc => Path imo